### PR TITLE
FEATURE: CL-9217 Transform MD object and execute

### PR DESCRIPTION
**Pull request by akloboucnik**

### CODE REVIEW

**Comment by yenkins-admin** [comment]
Test FAILed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/58/

**Comment by yenkins-admin** [comment]
Test PASSed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/59/

**Comment by yenkins-admin** [comment]
Test PASSed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/60/

**Comment by yenkins-admin** [comment]
Test PASSed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/61/

**Comment by yenkins-admin** [comment]
Test PASSed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/63/

**Comment by yenkins-admin** [comment]
Test FAILed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/64/

**Comment by yenkins-admin** [comment]
Test PASSed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/65/

File: src/execution.js [hunk 1]
```diff
@@ -24,24 +34,29 @@ import { ajax, post } from './xhr';
...
-    // enrich configuration with supported properties such as
-    // where clause with query-like filters or execution context filters
-    const config = executionConfiguration || {};
-    ['filters', 'where', 'orderBy', 'definitions'].forEach(function assignProperties(property) {
-        if (config[property]) {
-            request.execution[property] = config[property];
-        }
-    });
+    if (isPlainObject(elements)) {
+        request = elements;
```
Comment (bredy) [inline src/execution.js]: This is little bit strange - executionConfiguration is totally ignored.

File: src/execution.js [hunk 2]
```diff
@@ -86,3 +101,92 @@ export function getData(projectId, elements, executionConfiguration) {
     return d.promise();
 }
 
+const getFilterExpression = listAttributeFilter => {
+    const attributeUri = get(listAttributeFilter, 'listAttributeFilter.attribute');
+    const elements = get(listAttributeFilter, 'listAttributeFilter.default.attributeElements', []);
```
Comment (bredy) [inline src/execution.js]: What if elements/elementsForQuery are empty ? We'll get invalid MAQL condition [attribute uri] NOT IN/IN ().

File: src/execution.js [hunk 3]
```diff
@@ -86,3 +101,92 @@ export function getData(projectId, elements, executionConfiguration) {
...
+    const elementsForQuery = map(elements, e => `[${e}]`);
+    const negative = get(listAttributeFilter, 'listAttributeFilter.default.negativeSelection') ? 'NOT ' : '';
+
+    return `[${attributeUri}] ${negative}IN (${elementsForQuery.join(',')})`;
+};
+
+const getFactMetricExpression = factMetric => {
+    const aggregation = get(factMetric, 'aggregation', '').toUpperCase();
+    const identifier = get(factMetric, 'objectUri');
```
Comment (bredy) [inline src/execution.js]: Please rename to fact/metricUri

File: src/execution.js [hunk 4]
```diff
@@ -86,3 +101,92 @@ export function getData(projectId, elements, executionConfiguration) {
...
+
+    return { element, definition };
+};
+
+const categoryToElement = c => {
+    return { element: get(c, 'displayForm') };
+};
+
+const attributeFilterToWhere = f => {
+    const attributeUri = get(f, 'listAttributeFilter.displayForm');
```
Comment (bredy) [inline src/execution.js]: attributeUri or displayFormUri ?

File: src/execution.js (hunk 2 above)
Comment (akloboucnik) [inline src/execution.js]: Filter with empty elements will not be stored in MD - see https://github.com/gooddata/gdc-analytical-designer/pull/104/commits/89d82e266f18cd716296262ebfcbbf97ca3ff309

File: src/execution.js [hunk 5]
```diff
@@ -24,24 +34,29 @@ import { ajax, post } from './xhr';
...
+    } else {
+        // Create request and result structures
+        request = {
+            execution: {
+                columns: elements
+            }
+        };
+        // enrich configuration with supported properties such as
+        // where clause with query-like filters or execution context filters
+        const config = executionConfiguration || {};
```
Comment (bredy) [inline src/execution.js]: this could be assign as executionConfiguration default parameter in method signature

File: src/execution.js (hunk 2 above)
Comment (bredy) [inline src/execution.js]: I suppose this lib should handle it event if there are no elements.

**Comment by yenkins-admin** [comment]
Test PASSed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/66/

File: src/execution.js (hunk 2 above)
Comment (akloboucnik) [inline src/execution.js]: Ok, I've added a test case - see https://github.com/gooddata/gooddata-js/pull/122/commits/bcee77e78cbe295b5533b4bc724ac1fa292bbb3b

**Comment by yenkins-admin** [comment]
Test PASSed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/67/

**Comment by yenkins-admin** [comment]
Test PASSed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/68/

**Comment by yenkins-admin** [comment]
Test PASSed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/69/

File: src/execution.js [hunk 6]
```diff
@@ -86,3 +94,94 @@ export function getData(projectId, elements, executionConfiguration) {
...
+    const columns = [];
+    let where = {};
+    const definitions = [];
+    factMetrics.forEach(({element, definition}) => {
+        columns.push(element);
+        definitions.push(definition);
+    });
+    metrics.forEach(({element}) => columns.push(element));
+    attributes.forEach(({element}) => columns.push(element));
+    attributeFilters.forEach(f => {
```
Comment (vaclavbohac) [inline src/execution.js]: Seems like reduce to me.

**Comment by yenkins-admin** [comment]
Test FAILed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/70/

**Comment by yenkins-admin** [comment]
Test PASSed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/71/

**Comment by yenkins-admin** [comment]
Test PASSed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/72/

**Comment by yenkins-admin** [comment]
Test PASSed.
Refer to this link for build results: https://ci.intgdc.com/job/gooddata-js-pull-request-builder/73/

**Pull request merged**